### ISSUE TITLE
twisted.words msn broken

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3394]
|<img alt="Joaquin's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Joaquin reported|
|-|-|
|Trac ID|trac#3394|
|Type|defect|
|Created|2008-08-22 20:01:18Z|
|Branch|https://github.com/twisted/twisted/tree/msnp8-cvr0-3394|

presumably msn protocol has changed

when running msn_example.py

I get this:
```
2008/08/22 17:29 +0200 [-] Log opened.
2008/08/22 17:29 +0200 [-] Starting factory
<twisted.internet.protocol.ClientFactory instance at 0x83d552c>
2008/08/22 17:29 +0200 [Dispatch,client] Error in line: VER 1 MSNP8 (Invalid version response)
2008/08/22 17:29 +0200 [Dispatch,client] Stopping factory
<twisted.internet.protocol.ClientFactory instance at 0x83d552c>
```
1 or 2 months ago, it was working perfectly fine.

So I suspect that something has changed on messenger.hotmail.com side

bonus: this two-week old bug (closed now) seems to be related and claims to have a solution:
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=493908

    ''[...] their response to the CVR command has changed [...]''

Hopefully, it might be only that...

<details><summary>Searchable metadata</summary>

```
trac-id__3394 3394
type__defect defect
reporter__Joaquin Joaquin
priority__normal normal
milestone__ 
branch__branches_msnp8_cvr0_3394 branches/msnp8-cvr0-3394
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1219435278000000 1219435278000000
changetime__1220273922000000 1220273922000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [24622]) Branching to 'msnp8-cvr0-3394'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

`DispatchClient` and `NotificationClient` both handle VER messages without `CRV0` in them now.  I ran msn_example.py but I don't know what it was supposed to do.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph set owner to @exarkun|
|-|-|

Hrm.  It would be nice to have better QA in place for MSN, but based on the description here it looks like it fixes the problem.  Better test coverage is good in any case.  Waiting to set that up would probably be counterproductive given that this code is already broken against the only server in the world where it matters - merge away.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [24660]) Merge msnp8-cvr0-3394

Author: exarkun
Reviewer: glyph
Fixes: [#3394](https://github.com/twisted/twisted/issues/3394)

Change the MSN protocol implementation so that it does not require
the `CVR0` protocol to be included in the `VER` command.  The value
was never used for anything, although it was always previously
sent by the server.  Now it is no longer sent, so requiring it
only breaks the client.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
